ui/app/AppLayouts/Wallet/adaptors/TokenSelectorViewAdaptor.qml

```diff
@@ -0,0 +1,166 @@
+      - name: string -> user visible token name (e.g. "Ethereum")
+      - symbol: string -> user visible token symbol (e.g. "ETH")
+      - decimals: int -> number of decimal places
+      - communityId: string -> optional; ID of the community this token belongs to, if any
+      - marketDetails: var -> object containing props like `currencyPrice` for the computed values below
+      - balances: submodel -> [ chainId:int, account:string, balance:BigIntString, iconUrl:string ]
+
+      Computed values:
+      - currencyBalance: double (e.g. `1000.42` in user's fiat currency)
+      - currencyBalanceAsString: string (e.g. "1 000,42 CZK" formatted as a string according to the user's locale)
+      - balanceAsString: string (`1.42` formatted as e.g. "1,42" in user's locale)
+    */
+
+    // input API
+    required property var assetsModel
+    required property var flatNetworksModel
+    required property string currentCurrency // CurrenciesStore.currentCurrency, e.g. "USD"
+
+    // optional filter properties; empty/default values means no filtering
+    property var enabledChainIds: []
+    property string accountAddress
+    property bool showCommunityAssets
+    property string searchString
+
+    // output model
+    readonly property SortFilterProxyModel outputAssetsModel: SortFilterProxyModel {
+        sourceModel: assetsObjectProxyModel
+
+        filters: [
+            AnyOf {
+                RegExpFilter {
+                    roleName: "name"
+                    pattern: root.searchString
+                    caseSensitivity: Qt.CaseInsensitive
+                }
+                RegExpFilter {
+                    roleName: "symbol"
+                    pattern: root.searchString
+                    caseSensitivity: Qt.CaseInsensitive
+                }
+            },
+            ValueFilter {
+                roleName: "communityId"
+                value: ""
+                enabled: !root.showCommunityAssets
+            }
+        ]
+
+        // FIXME optionally sort/filter by wallet controller as well
+        sorters: [
+            RoleSorter {
+                roleName: "currencyBalance"
+                sortOrder: Qt.DescendingOrder
+            }
+        ]
+    }
+
+    // internals
+    ObjectProxyModel {
+        id: assetsObjectProxyModel
+        sourceModel: root.assetsModel
+
+        delegate: SortFilterProxyModel {
+            id: delegateRoot
+
+            // properties exposed as roles to the top-level model
+            readonly property int decimals: model.decimals
+            readonly property double currentBalance: aggregator.value
+            readonly property double currencyBalance: {
+                if (!!model.marketDetails) {
+                    return currentBalance * model.marketDetails.currencyPrice.amount
+                }
+                return 0
+            }
+            readonly property string currencyBalanceAsString:
```

Comment:
It's a bit better because it provides correct behavior, but the displayDecimals and formatting rule in general is constant for all entries in the model. So there is no need to expect additional field in the model per row (model.marketDetails.currencyPrice.displayDecimals) and expose additional role. Something changing across entries should be passed via the model, fixed stuff can be passed via a single property, directly to the view, no need to go via adaptor. For that reason formatting function passed to the view is good option here.

Reply:
It is not, check my answer above

Reply:
I mean the same value for all items in the model. From perspective of the model it's just single param.